### PR TITLE
Fix tests failings for TwoPointBVPFunction

### PR DESCRIPTION
**Pull request by ErikQQY**
This should fix the tests failings in DiffEqBase.jl https://github.com/SciML/DiffEqBase.jl/pull/940

@ChrisRackauckas @avik-pal

### CODE REVIEW

File: src/problems/bvp_problems.jl [hunk 1]
```diff
@@ -119,25 +119,25 @@ struct BVProblem{uType, tType, isinplace, P, F, BF, PT, K} <:
         else
             @assert prob_type === problem_type "This indicates incorrect problem type specification! Users should never pass in `problem_type` kwarg, this exists exclusively for internal use."
         end
-        return new{typeof(u0), typeof(_tspan), iip, typeof(p), typeof(f), typeof(bc),
-            typeof(problem_type), typeof(kwargs)}(f, bc, u0, _tspan, p, problem_type,
+        return new{typeof(u0), typeof(_tspan), iip, typeof(p), typeof(f), typeof(f.bc),
```
Comment (ChrisRackauckas) [inline src/problems/bvp_problems.jl:122]: Why is the bc out here at all? Isn't it in the BVPFunction?

Comment (ChrisRackauckas) [reply src/problems/bvp_problems.jl:122]: @avik-pal is it actually needed?

Comment (avik-pal) [reply src/problems/bvp_problems.jl:122]: The `bc` was kept outside similar to the SDEProblem interface carrying around the `g` outside (I think?)

Comment (ChrisRackauckas) [reply src/problems/bvp_problems.jl:122]: That was just fixed in the v2.0. I thought this was fixed in the v2.0 as well 😅. Fix and slip it in ASAP.

Comment (ErikQQY) [reply src/problems/bvp_problems.jl:122]: The  modifications in ```BVProblem``` is similiar to ```SDEProblem```, just try to make them consistent

Comment (avik-pal) [reply src/problems/bvp_problems.jl:122]: SDEProblem is still carrying around `g` https://github.com/SciML/SciMLBase.jl/blob/75b192581832ae89231df0d51a0fe9624e61470e/src/problems/sde_problems.jl#L88-L89

Comment (ErikQQY) [reply src/problems/bvp_problems.jl:122]: Yeah, ```SDEProblem``` is still carrying ```g```, so in this way I think we can still use ```prob.g``` to directly access ```g```

**Comment by ChrisRackauckas** [comment]
@avik-pal is this still needed after https://github.com/SciML/SciMLBase.jl/pull/518 ?

**Comment by ChrisRackauckas** [comment]
This PR is now out of date because of the changes in https://github.com/SciML/SciMLBase.jl/pull/518, so closing, but a similar PR to the new function form is required.

**Pull request closed without merging**